portal/academy/services.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import random
+from itertools import zip_longest
+from io import StringIO
+
+from django.contrib.contenttypes.models import ContentType
+
+from portal.hackathons.models import Hackathon
+from portal.hackathons.models import Attendance
+from portal.users.models import User
+
+
+logger = logging.getLogger(__name__)
+
+
+def check_graduation_status(user: User):
+    """
+    Check graduation eligibility of student given their attendance in hackathons:
+
+    - if student missed the first hackathon, they can not not graduate
+    - if student has missed more than one hackathon, they can not not graduate
+    - otherwise the student can still graduate
+
+    """
+    logger.info(f"Checking graduation status for {user.name}")
```

Comment:
user.username is what is usually used to identify the user, it's more useful if we have to check logs.

Reply:
Thanks! done :)